test/requests/volume/quota_tests.rb

```diff
@@ -0,0 +1,40 @@
+require 'test_helper'
+
+describe "Fog::Volume[:openstack] | quota requests" do
+  before do
+    @volume = Fog::Volume[:openstack]
+    @tenant_id = Fog::Compute[:openstack].list_tenants.body['tenants'].first['id']
+    @quota_set_format = {
+      'volumes'   => Fog::Nullable::Integer,
+      'gigabytes' => Fog::Nullable::Integer,
+      'snapshots' => Fog::Nullable::Integer,
+      'id'        => String
+    }
+    @quota = @volume.get_quota(@tenant_id).body['quota_set']
+  end
+
+  describe "success" do
+    it "#get_quota_defaults" do
+      @volume.get_quota_defaults(@tenant_id).body.
+        must_match_schema('quota_set' => @quota_set_format)
+    end
+
+    it "#get_quota" do
+      @quota.must_match_schema(@quota_set_format)
+    end
+
+    it "updates quota" do
```

Comment:
Thanks for the detailed explanation. I agree the break down of describe/it can/must be improved for better readability. Again this is legacy code and the initial target is to port it to Minitest.
Regarding the specific "updates quota" I effectively don't see the point of asserting #update_quota twice. Therefore I remove the second one.